rfc/0078-zone-aware-replica-reads.md

```diff
@@ -0,0 +1,352 @@
+# Meta
+
+| Field          | Value                    |
+|----------------|--------------------------|
+| RFC Name       | Zone-Aware Replica Reads |
+| RFC ID         | 78                       |
+| Start Date     | 2024-04-25               |
+| Owner          | Sergey Avseyev           |
+| Current Status | DRAFT                    |
+| Revision       | #1                       |
+
+# Summary
+
+This document describes changes that SDKs must implement to optimize costs of
+network traffic when the server is being deployed across multiple availability
+zones.
+
+# Motivation
+
+Existing SDKs do not take into account node availability groups of the nodes in
+the cluster. In particular, when performing `getAllReplicas` operations, the SDK
+would send `GET (0x00)` request to active vBucket and `GET_REPLICA (0x83)` to
+each of the replicas. This approach will make reading from replicas expensive
+operations if nodes of the cluster deployed in different availability zones
+(AZ).
+
+# Use Cases
+
+## No Preference
+
+This is current behaviour, where the SDK does not take into account server
+groups.
+
+![No Prefrence](figures/0078-case-1-no-preference.svg)
+
+## Selected Server Group
+
+In this case, all reads will be done only from the selected server group. It
+is the cheapest solution, although it reduces chances of getting data from
+replicas (if the server groups are unbalanced).
+
+![Selected Server Group](figures/0078-case-2-local-only.svg)
+
+## Selected Server Group or All Available
+
+This strategy allows using all available replicas, but only in case when the
```

Comment:
Is this meant to mean "only in the case where the local server group doesn't exist" or "only in the case where the local server group has no nodes for the needed replica"

Reply:
I have updated the text to explain that this third case is just a fallback from the third, to make sure that the SDK allows user to do it. Because in certain setups it is not possible to make group to cover all 1024 vBuckets (when number of copies is less than number of groups, e.g.: 1 active + 2 replicas < 4 groups).